src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -116,10 +128,14 @@ private IssuesSnapshot(Guid snapshotId, string projectName, string filePath, IEn
             this.ruleHelpLinkProvider = ruleHelpLinkProvider;
             this.issueMarkers = new List<IssueMarker>(issueMarkers);
             this.readonlyIssueMarkers = new ReadOnlyCollection<IssueMarker>(this.issueMarkers);
+
+            this.FilesInSnapshot = CalculateFilesInSnapshot(issueMarkers);
```

Comment:
Optimisation to avoid having to walk all of the issues/flows/locations to determine if there are any relevant files.

Reply:
Perhaps we should calculate a `Dictionary<string,IAnalysisIssueLocationVisualization` instead. `FilesInSnapshot` could then be Dictionary.Keys, and GetLocationsVizsForFile will be O(1).

Reply:
The dictionary would have to be string->list of locations.

As discussed offline, I'm only trying to optimise the most common case of taggers being able quickly work out that they don't need to do any work because the file they manage isn't affected.
If it turns out we need to optimise the lookup too then we can do that separately.